### PR TITLE
Set unique attempt for each cri contianer create

### PR DESCRIPTION
**Pull request by koct9i**
- Set unique attempt for each CRI container create

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: yt/yt/library/containers/cri/cri_executor.cpp [hunk 1]
```diff
@@ -351,6 +352,9 @@ class TCriExecutor
         {
             auto* metadata = config->mutable_metadata();
             metadata->set_name(ctSpec->Name);
+
+            // Set unique attempt for each newly created container.
+            metadata->set_attempt(Attempt_++);
```
Comment (Kontakter) [inline yt/yt/library/containers/cri/cri_executor.cpp:357]: What this attempt field actually mean in CRI?

For me attempt notion is usually used in retries. But here we have CriExecutor that is responsible for all requests to CRI

Comment (koct9i) [reply yt/yt/library/containers/cri/cri_executor.cpp:357]: "Attempt" is designed to be used for retries for sure.
But it's just a number appened to container name.

https://github.com/containerd/containerd/blob/main/pkg/cri/server/helpers.go#L128

Unique key for create is (namespace, pod, name, attempt)
it returns container id and all futher operations are done by id.

We could add randomness to container name with same result.
At the end of each job CRI executor list sand removes all contianers in the pod thus garbage should be collected.

**Comment by gritukan** [comment]
We usually start commit name from a capital letter, so "set" -> "Set".

Besides this, LGTM.

**Review by gritukan — APPROVED** [review]

**Comment by yalekseev** [comment]
Shipped!

File: yt/yt/library/containers/cri/cri_executor.cpp (hunk 1 above)
Comment (Kontakter) [reply yt/yt/library/containers/cri/cri_executor.cpp:367]: I would prefer to add randomness to the name, because the current solution is misleading.

**Pull request closed without merging**